src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,52 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<F, S> {
+    private F first;
+    private S second;
 
+    private Pair(F first, S second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public F getFirst() {
+        return first;
+    }
+
+    public S getSecond() {
+        return second;
+    }
+
+    public static Pair of(Object first, Object second) {
+        return new Pair(first, second);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+
+        if (obj == null) {
+            return false;
+        }
+
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj.getClass().equals(getClass())) {
+            Pair pair = (Pair) obj;
+
+            return (pair.first == null && first == null) || first.equals(pair.first)
```

Comment:
what happens on this row if pair.first =! null and first == null ?

Reply:
@RomanDubovskyi If pair.first != null and first == null, method will return false, because in this case first.equals(pair.first) will be false automatically.

Reply:
please check this situation with your current impl and let me know what would be the result 
```
Pair<String, String> pair = Pair.of(null, "2");
    Pair<String, String> pair2 = Pair.of("first", "second");
    System.out.println(pair.equals(pair2));
```